packages/block-renderer/src/hooks/use-pattern-inline-css.tsx

```diff
@@ -0,0 +1,29 @@
+import { shuffle } from '@automattic/js-utils';
+import { useMemo } from 'react';
+import { MAX_BLOG_POSTS } from '../constants';
+
+// Using this object to memoize the pattern as a workaround
+const memoPatternOrder: { [ key: string ]: string } = {};
+
+const usePatternInlineCss = ( patternId: string ) => {
+	// FIXME: Why this useMemo is not working?
+	return useMemo( () => {
+		let inlineCss = memoPatternOrder[ patternId ] || '';
+
+		if ( inlineCss ) {
+			return inlineCss;
+		}
+
+		shuffle( [ ...Array( MAX_BLOG_POSTS ).keys() ] ).forEach( ( order, index ) => {
+			const childIndex = index + 1;
+			const shuffledOrder = order + 1;
+			inlineCss += `.is-layout-grid .wp-block-post:nth-child(${ childIndex }) { order: ${ shuffledOrder }; }`;
```

Comment:
Nit: I think it would be better to target the direct children of the `.is-layout-grid` directly using `>`:

```
.is-layout-grid > .wp-block-post:nth-child ...
```

or maybe even better for more readability and self-documentation:

```
ul.is-layout-grid > li.wp-block-post:nth-child ...
```

Reply:
I've added the `>` to target direct children 👍 

I'm not sure about using `ul` and `li` here because they could change and break the logic. Plus, for me, they don't make the CSS more readable or document anything relevant for the assembler. Anyway, thanks for the idea 🙂

Reply:
Fair enough, I was thinking that it will make it clearer that we're manipulating list items in a list (not just random divs).

But I think you're right that it could change and can break the logic, so it's fine 😄